app/models/result.rb

```diff
@@ -59,7 +59,7 @@ def get_subtotal(user_visibility = :ta)
     unless marks.empty?
       assignment = submission.grouping.assignment
       assignment.get_criteria(user_visibility).each do |criterion|
-        mark = marks.find_by(markable_id: criterion.id)
+        mark = marks.find_by(markable_id: criterion.id, markable_type: criterion.class.to_s)
```

Comment:
Your first paragraph is correct.

The test that was failing is because the created flexible criterion isn't associated with the proper assignment.